### ISSUE TITLE
Recent changes made Scene switching not possible

### ISSUE THREAD

**Issue by DoctorWhoof** [blitz-research/monkey2#282]
This DebugAssert line was recently added to "scene.monkey2" in the dev branch:
```
Function SetCurrent( scene:Scene )
	DebugAssert( Not _current,"Scene.Current already set" )    '<---- This was added recently
	_current=scene
End
```

I used to be able to switch between scenes using SetCurrent, now I can't anymore, since it errors out if _current is already assigned. Is there a better way to change scenes, or was this result not intentional?

Thanks!

**Comment by blitz-research** [comment]
Wow, someone's actually using multiple scenes?!? I'll revert this back to
the way it was soon. Does your code still work with the Assert removed?

The truth is I'm still not quite clear on how this stuff should eventually
work. I don't like how ALL entities are just added to the 'DefaultScene'
(better name IMO) whether you like it or not. It would be nice to be able
to load/create entities and not have them attached to ANY scene, instead of
having to hide them if you want to create copies of them when the game gets
underway.

I would quite like people to have to explicitly add entities to a scene (or
have a Scene:RootEntity style property) and get rid of 'current' scene
altogether. Or at least offer the choice somehow. Still thinking about this
side of things anyway...there are pros and cons to every approach!

**Comment by DoctorWhoof** [comment]
Yeah, the Assert is the only thing preventing it from working.

I agree that having entities only exist in their scenes would be more efficient, although at the moment the current system works for me, since I'm not breaking anything into "levels" - I like the single, contiguous world design style.

The main reason I need to switch to a new scene is just when I "reload" the scene. The current scene is discarded, I call OnDestroy() on all game objects to ensure all stacks are cleared, etc, then create a fresh new scene and load new entities from the .json file so I can see the recent changes I've made to it.

> I would quite like people to have to explicitly add entities to a scene

I like that approach too, "manually" creating a scene, then specifying the scene an entity belongs to when it's created. That said, it's very helpful to somehow allow an entity to exist in more than a single scene - let's say a player entity that preserves its attributes across multiple levels.

Thanks!

**Comment by blitz-research** [comment]
So if I add a Scene,Clear() you wouldn't need multiple scenes?

There is already a Scene.DestroyAllEntities() in there, a Scene.Clear()
would probably be the same thing plus setting any skyboxes etc back to
default.

**Comment by DoctorWhoof** [comment]
Yes, I'm already calling Scene.DestroyAllEntities, after I go through each GameObject and make sure nothing is gonna be left behind. Scene.Clear() would probably work in my case, since the default scene would behave like a brand new one after it.

Side note: I'm still using GameObjects that "hold" the Entities, even after you added Components to Entities. I haven't had time to try to extend the current entities into what I need yet (More OnStuff() events, Json serialization that converts things like vectors and colors into arrays, etc.).

**Comment by blitz-research** [comment]
> I'm still using GameObjects that "hold" the Entities, even after you
added Components to Entities.

Probably sensible - I'm still really just playing around here, but I do
like where it's going.
